src/lib.rs

```diff
@@ -95,23 +94,36 @@ impl<E: Env> Ctap<E> {
         &mut self,
         packet: &HidPacket,
         transport: Transport,
-        now: CtapInstant,
     ) -> HidPacketIterator {
         match transport {
             Transport::MainHid => {
                 self.hid
-                    .process_hid_packet(&mut self.env, packet, now, &mut self.state)
+                    .process_hid_packet(&mut self.env, packet, &mut self.state)
             }
             #[cfg(feature = "vendor_hid")]
             Transport::VendorHid => {
                 self.vendor_hid
-                    .process_hid_packet(&mut self.env, packet, now, &mut self.state)
+                    .process_hid_packet(&mut self.env, packet, &mut self.state)
             }
         }
     }
 
-    pub fn update_timeouts(&mut self, now: CtapInstant) {
-        self.state.update_timeouts(now);
-        self.hid.update_wink_timeout(now);
+    pub fn update_timeouts(&mut self) {
+        self.env.clock().tickle();
+        self.state.update_timeouts(&mut self.env);
```

Comment:
Do we agree that in theory we don't need to update the timeouts here and can instead just `check_timer` when we want to use them?

Reply:
I agree. I did that for wink since the change was obvious. For the other two functions called here, it would have needed some more reshulffing. In the case of `StatefulPermission`, the reason my bad design: The function that checks the timer resets the allowed command on timeout, and therefore relies on us correctly calling `clear_timer`. Not hard to do, but I didn't want to make this PR bigger than necessary.

For `ClientPin` the problem is similar. Here, the implementation mostly follows the specification word by word. Not necessariy the best code design, but easy to reason about. Not hard to fix, but can be a follow-up PR.

Reply:
Sounds good, no need to take action in this PR.